packages/dds/tree/src/test/shared-tree/summaryBenchmarkSizes.ts

```diff
@@ -0,0 +1,173 @@
+        const tree = provider.trees[0];
+
+        const { summary } = tree.getAttachSummary();
+        const summaryString = JSON.stringify(summary);
+        const summarySize = IsoBuffer.from(summaryString).byteLength;
+        assert(summarySize < 1000);
+    });
+    it("with 1 inserted node.", async () => {
+        const summarySize = await getInsertsSummarySize(1, 0, false);
+        assert(summarySize < 2000);
+    });
+    it("with 10 inserted nodes width-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(10, 0, false);
+        assert(summarySize < 3000);
+    });
+    it("with 100 inserted nodes width-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(100, 0, false);
+        assert(summarySize < 20000);
+    });
+    it("with 10 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(10, 0, true);
+        assert(summarySize < 3000);
+    });
+    it("with 100 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(100, 0, true);
+        assert(summarySize < 20000);
+    });
+    it("rejected for 1000 inserts depth wise", async () => {
+        await assert.rejects(getInsertsSummarySize(1000, 0, true), { message: "BatchTooLarge" });
+    });
+});
+
+/**
+ * Inserts a single node under the root of the tree with the given value.
+ */
+function setTestValue(tree: ISharedTree, value: TreeValue, index: number): void {
+    // Apply an edit to the tree which inserts a node with a value
+    tree.runTransaction((forest, editor) => {
+        const writeCursor = singleTextCursor({ type: brand("TestValue"), value });
+        const field = editor.sequenceField(undefined, rootFieldKeySymbol);
+        field.insert(index, writeCursor);
+
+        return TransactionResult.Apply;
+    });
+}
+
+function setTestValueOnPath(tree: ISharedTree, value: TreeValue, path: PlacePath): void {
+    // Apply an edit to the tree which inserts a node with a value.
+    tree.runTransaction((forest, editor) => {
+        const writeCursor = singleTextCursor({ type: brand("TestValue"), value });
+        const field = editor.sequenceField(path, rootFieldKeySymbol);
+        field.insert(0, writeCursor);
+
+        return TransactionResult.Apply;
+    });
+}
+
+function setTestValuesWide(tree: ISharedTree, seed: number, numNodes: number): void {
+    const random = makeRandom(seed);
+    for (let j = 0; j < numNodes; j++) {
+        setTestValue(tree, random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER), j);
+    }
+}
+
+/**
+ *
+ * @param numNodes - number of nodes you would like to insert
+ * @param seed - seed to randomly generate values for the inserts
+ * @param depth - boolean to specify inserting nodes depth wise or width wise
+ * @returns the byte size of the tree's summary
+ */
+export async function getInsertsSummarySize(
+    numNodes: number,
```

Comment:
We try to avoid abbreviations in names. "numberOfNodes" would be more clear.

Reply:
updated parameter name